ccmlib/node.py

```diff
@@ -1254,22 +1257,34 @@ def _update_pid(self, process):
             raise NodeError('Problem starting node %s' % self.name, process)
         self.__update_status()
 
-    def __gather_sstables(self, datafile=None, keyspace=None, columnfamilies=None):
-        datafiles = []
+    def __gather_sstables(self, datafiles=None, keyspace=None, columnfamilies=None):
+        files = []
         if keyspace is None:
-            for k in self.list_keyspaces():
-                datafiles = datafiles + self.get_sstables(k, "")
-        elif datafile is None:
+            if datafiles:
+                for datafile in datafiles:
+                    if not os.path.isabs(datafile):
+                        datafile = os.path.join(os.getcwd(), datafile)
```

Comment:
Why os.getcwd()? It seems that you're adding the functionality to call sstablesplit on sstables that are not inside the C* data directory here, which is fine, but i don't think that should be in __gather_sstables. I would prefer to isolate the behavior of this function to gathering the paths of sstables inside the C* data directory based on keyspace, CF, etc.